stac_fastapi/types/stac_fastapi/types/aggregation.py

```diff
@@ -0,0 +1,45 @@
+"""Aggregation Extension types."""
+
+import sys
+from typing import Any, Dict, List, Literal, Optional, Union
+
+from pydantic import Field
+
+from stac_fastapi.types.rfc3339 import DateTimeType
+
+# Avoids a Pydantic error:
+# TypeError: You should use `typing_extensions.TypedDict` instead of
+# `typing.TypedDict` with Python < 3.12.0.  Without it, there is no way to
+# differentiate required and optional fields when subclassed.
+if sys.version_info < (3, 12, 0):
+    from typing_extensions import TypedDict
+else:
+    from typing import TypedDict
```

Comment:
We can import TypedDict directly from typing extension

- (as usual with `typing extension`) we can import type from the module, and it will automatically choose between the native type or a mock one: https://github.com/python/typing_extensions/blob/main/src/typing_extensions.py#L870-L890

- `typing-extensions` is a requirement of `fastapi` so we can almost safely import it

Reply:
Are you saying that only the `from typing_extensions import TypedDict` line is needed because the python version condition is in the `typing_extension` library?

If so we can update here as well. https://github.com/stac-utils/stac-fastapi/blob/main/stac_fastapi/types/stac_fastapi/types/stac.py#L12

Reply:
Updated in latest commti